server/service/mdm.go

```diff
@@ -1980,3 +1980,138 @@ func (svc *Service) UpdateMDMDiskEncryption(ctx context.Context, teamID *uint, e
+	}
+
+	return resendHostMDMProfileResponse{}, nil
+}
+
+func (svc *Service) ResendHostMDMProfile(ctx context.Context, hostID uint, profileUUID string) error {
+	// first we perform a perform basic authz check, we use selective list action to include gitops users
+	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionSelectiveList); err != nil {
+		return ctxerr.Wrap(ctx, err)
+	}
+
+	host, err := svc.ds.HostLite(ctx, hostID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err)
+	}
+
+	// now we can do a specific authz check based on team id of the host before proceeding
+	if err := svc.authz.Authorize(ctx, &fleet.MDMConfigProfileAuthz{TeamID: host.TeamID}, fleet.ActionWrite); err != nil {
+		return ctxerr.Wrap(ctx, err)
+	}
+
+	// TODO: prescreen host platform against profile prefix?
+
+	var profileTeamID *uint
+	var profileName string
+	switch {
+	case strings.HasPrefix(profileUUID, fleet.MDMAppleProfileUUIDPrefix), strings.HasPrefix(profileUUID, fleet.MDMAppleDeclarationUUIDPrefix):
+		if err := svc.VerifyMDMAppleConfigured(ctx); err != nil {
+			return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("profile_uuid", fleet.AppleMDMNotConfiguredMessage).WithStatus(http.StatusBadRequest), "check apple mdm enabled")
+		}
+		prof, err := svc.ds.GetMDMAppleConfigProfile(ctx, profileUUID)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "getting apple config profile")
+		}
+		profileTeamID = prof.TeamID
+		profileName = prof.Name
+
+	case strings.HasPrefix(profileUUID, fleet.MDMAppleDeclarationUUIDPrefix):
+		if err := svc.VerifyMDMAppleConfigured(ctx); err != nil {
+			return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("profile_uuid", fleet.AppleMDMNotConfiguredMessage).WithStatus(http.StatusBadRequest), "check apple mdm enabled")
+		}
+		decl, err := svc.ds.GetMDMAppleDeclaration(ctx, profileUUID)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "getting apple declaration")
+		}
+		profileTeamID = decl.TeamID
+		profileName = decl.Name
+
+	case strings.HasPrefix(profileUUID, fleet.MDMWindowsProfileUUIDPrefix):
+		if err := svc.VerifyMDMWindowsConfigured(ctx); err != nil {
+			return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("profile_uuid", fleet.WindowsMDMNotConfiguredMessage).WithStatus(http.StatusBadRequest), "check windows mdm enabled")
+		}
+		prof, err := svc.ds.GetMDMWindowsConfigProfile(ctx, profileUUID)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "getting windows config profile")
+		}
+		profileTeamID = prof.TeamID
+		profileName = prof.Name
+
+	default:
+		// TODO: confirm error message?
+		return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("profile_uuid", "Invalid profile UUID prefix."))
+	}
+
+	// check again based on team id of profile before we proceeding
+	if err := svc.authz.Authorize(ctx, &fleet.MDMConfigProfileAuthz{TeamID: profileTeamID}, fleet.ActionWrite); err != nil {
+		return ctxerr.Wrap(ctx, err)
+	}
+
+	// TODO: specs say to screen disk encryption profile? what about other reserved profiles?
```

Comment:
@marko-lisica another sanity check, we also have:

- Fleetd profile
- Windows OS Updates profile

Reply:
@roperzh We don't want to display Windows OS updates in the profile list, since it has separate UI. So user shouldn't be able to see that one or resend.

What do you mean by Fleetd profile? I see `Fleetd configuration` profile in the profile list (user should be able to resend this profile). 

I just realized that I've didn't remove [dev note](https://www.figma.com/file/CtUf2RRqK4eEOTv6W5YjUD/%2317314-Resend-a-configuration-profile?type=design&node-id=324-522&mode=design&t=CFdeqrUJAZdQ35z5-11) about disk encryption profile. We decided to let users resend the disk encryption profile since it [won't make end-users encrypt again](https://fleetdm.slack.com/archives/C03C41L5YEL/p1711396883606039). I hope this isn't huge change, to remove this limitation?

Reply:
@marko-lisica thanks!

> We don't want to display Windows OS updates in the profile list, since it has separate UI. So user shouldn't be able to see that one or resend.

sorry for not being clear on that! it might be limited in the UI, but an API user can still request any profile to be resent.

> I just realized that I've didn't remove [dev note](https://www.figma.com/file/CtUf2RRqK4eEOTv6W5YjUD/%2317314-Resend-a-configuration-profile?type=design&node-id=324-522&mode=design&t=CFdeqrUJAZdQ35z5-11) about disk encryption profile. We decided to let users resend the disk encryption profile since it [won't make end-users encrypt again](https://fleetdm.slack.com/archives/C03C41L5YEL/p1711396883606039). I hope this isn't huge change, to remove this limitation?

I don't think that will be a problem, in that case my hunch is to allow the Win OS updates as well?

Reply:
@roperzh As far as I know OS update profile hasn't surfaced to users (via API as well). How would they resend it? I still think we should prevent that if it's possible to get uuid of that profile somehow.

Reply:
@marko-lisica sending the profile again won't have any side effects AFAIK (just a couple of `Replace`s) that's why I was thinking we should apply the same rationale as for FileVault.

there are many ways to get the UUID, eg: by viewing the command results via fleetctl.

Reply:
That's probbly ok. If they resend profile, it won't change anything because those values are set already. I'll bring this to design review for emergency drafting and let you know what we decided.

Reply:
Thanks!

Reply:
from Marko/Noah: let's remove the restriction for FileVault, it's also OK for the user to re-send any of Fleet managed profiles.

Reply:
cc: @gillespi314 ☝️